.eslintrc.js

```diff
@@ -0,0 +1,36 @@
+module.exports = {
+  root: true,
+
+  env: {
+    node: true,
+  },
+
+  extends: [
+    'plugin:vue/recommended',
+    '@vue/airbnb',
+    '@vue/typescript',
+    'plugin:vue/recommended',
+    '@vue/typescript/recommended',
+  ],
+
+  plugins: [
+    'vuetify',
+  ],
+
+  rules: {
+    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
+    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
+    'max-len': ['off'],
+    'import/prefer-default-export': ['off'],
+    'no-underscore-dangle': ['off'],
+    'no-param-reassign': ['error', { props: false }],
```

Comment:
Do you need this line? It seems like this is the default value for `no-param-reassign`.

Reply:
Yes, without it there are a lot of errors. I'm not sure if it's default in a different version of eslint, and not ours

Reply:
Which errors do you man? This rule seems to say that it will *cause* errors if you are reassigning parameters or their properties.

Reply:
You said this was the default, but when I remove this line, I get 25 errors like this:

```
error: Assignment to property of function parameter 'd' (no-param-reassign) at src/lib/multinet.ts:73:51:
  71 | // eslint-disable-next-line  @typescript-eslint/no-explicit-any
  72 | function _defineNeighbors(nodes: any[], links: any[]) {
> 73 |   nodes.forEach((d: { neighbors: string[] }) => { d.neighbors = []; });
     |                                                   ^
  74 | 
  75 |   links.forEach((link) => {
  76 |     nodes.filter((d: Node) => d._id === link._from)[0].neighbors.push(link._to);
```

Reply:
Ok I see. Same as above, I'd put in a comment explaining that you need this rule override in order to allow D3-style mutating callbacks.

As a general point, disabling a rule just to suppress errors (with no further justification) is in general a no-go. In this case, by contrast, we do have a good reason to allow parameter property mutation--namely that it's a very common pattern for D3. Even then, I'd prefer in general to use inline disable directives to accommodate things like D3 (so that in non-D3 contexts you aren't doing this), but in this case I will leave it up to you. As above, I just would request a comment in the eslint rules explaining why we're doing it (in particular, so that we don't blindly copy this rule to other projects where it may not apply).

Reply:
Fixed in 831e857